collectors/prometheus.go

```diff
@@ -94,11 +96,21 @@ func NewPrometheusExporter(cfg *PrometheusConfig, lnd *lndclient.LndServices,
 
 	htlcMonitor := newHtlcMonitor(lnd.Router, errChan)
 
+	// Setup signalling so SIGTERM || SIGINT will stop the cache goroutine
+	// from ChannelsCollector
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+	chanCollector := NewChannelsCollector(
+		lnd.Client, errChan, monitoringCfg,
+	)
+	go func() {
+		<-sigs
+		close(chanCollector.quit)
```

Comment:
nit: would be nicer to inject a `quit` channel to `NewChannelsCollector` as a dependency.